primo/data_parser/well_data.py

```diff
@@ -915,6 +924,92 @@ def compute_priority_scores(self, impact_metrics: ImpactMetrics):
                 )
                 raise_exception(msg, ValueError)
 
+    def _process_dac_data(self):
+        """
+        processes the dac data
+        """
+        self._append_fed_dac_data()
+        return
+
+    def _compute_well_count_score(self, metric: Metric, metrics: ImpactMetrics):
+        """
+        process well count data
+        """
+        operator_name = self._col_names.operator_name
+        weight = metrics.well_count.effective_weight
+        metric.data_col_name = "Owner Well-Count"
+        self.data[metric.data_col_name] = self.data.groupby(operator_name)[
+            operator_name
+        ].transform("size")
+        self.data[metric.score_col_name] = round(
+            weight
+            * self.data[metric.data_col_name].apply(lambda x: math.e ** ((1 - x) / 10)),
+            4,
+        )
+
+    def set_impact_and_efficiency_metrics(
+        self,
+        impact_metrics: ImpactMetrics = None,
+        efficiency_metrics: EfficiencyMetrics = None,
+    ):
+        """
+        Validates and processes data for the impact and efficiency metrics.
+
+        Parameters
+        ----------
+        impact_metrics : ImpactMetrics
+            impact metrics object to be used for prioritization
+        efficiency_metrics : EfficiencyMetrics
+            efficiency metrics object to be used for efficiency score computation
+        """
+        if impact_metrics is not None:
+            self.config.impact_metrics = impact_metrics
+            self._set_metric(self.config.impact_metrics)
+        if efficiency_metrics is not None:
+            self.config.efficiency_metrics = efficiency_metrics
+            self._set_metric(self.config.efficiency_metrics)
+
+    def compute_priority_scores(self):
+        """
+        Computes scores for all metrics/submetrics (supported and custom metrics) and
+        the total priority score. This method must be called after processing the
+        data for custom metrics (if any).
+
+        Parameters
+        ----------
+        impact_metrics : ImpactMetrics
+            Object containing impact metrics and their weights
+        """
+        # Check the validity of impact metrics before calculating priority score
+        im_mt = self.config.impact_metrics
+        im_mt.check_validity()
+
+        # Check if all the required columns for supported metrics are specified
+        # If yes, register the name of the column containing the data in the
+        # data_col_name attribute
+        # TODO: Combine the check_columns_available method with this method.
+        # TODO: Check fill data consistency for ann_gas_production and
+        # ann_oil_production. See _categorize_gas_oil_wells method for details.
+        self._col_names.check_columns_available(im_mt)
```

Comment:
We can remove lines 984, 985, and 993, because these checks are performed in `_set_metric` and `_process_data` methods when the metric objects are registered.

Reply:
Good catch.